Fix tf2_ros test errors across all platforms

evshary opened the pull request:
While running [CI test](http://build.ros2.org/job/Eci__nightly-cyclonedds_ubuntu_bionic_amd64/lastCompletedBuild/testReport/tf2_ros.test.py_test.test_buffer/TestBuffer/test_can_transform_valid_transform/), there are some error messages like this

```
AttributeError: module 'rclpy.timer' has no attribute 'Clock'
```

I think the root cause is the wrong path of class Clock. It should be `rclpy.clock.Clock()`.

evshary committed:
class Clock is in clock not timer.

Signed-off-by: evshary <evshary@gmail.com>

clalancette committed:
More test fixes for tf2_ros python.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette commented:
Thanks for the fix.  It looks like I had a typo when running CI on https://github.com/ros2/geometry2/pull/99 code yesterday, so all of the tests didn't run.  I've added a couple more fixes here, and I'll now run CI to see if this fixes most of the problems:

Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=8155)](http://ci.ros2.org/job/ci_linux/8155/)

clalancette committed:
Remove unnecessary and blacklisted actionlib_msgs dependency.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette commented:
Linux is fixed with the changes here.  There is one more error coming from Windows Debug that I'm working on.

clalancette committed:
Minor fix to use CMake variable.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette committed:
Make BufferInterface destructor virtual.

Since it is a virtual class, we want to make sure that the
destructor is virtual.  This fixes a warning from clang.

Signed-off-by: Chris Lalancette <clalancette@openrobotics.org>

clalancette commented:
Here's a CI on everything except for Windows Debug (I may end up addressing that in a separate PR, since I want to get these changes in for everything else):

* Linux [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=8157)](http://ci.ros2.org/job/ci_linux/8157/)
* Linux-aarch64 [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux-aarch64&build=4140)](http://ci.ros2.org/job/ci_linux-aarch64/4140/)
* macOS [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_osx&build=6643)](http://ci.ros2.org/job/ci_osx/6643/)
* Windows [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_windows&build=8037)](http://ci.ros2.org/job/ci_windows/8037/)
* Linux clang [![Build Status](http://ci.ros2.org/buildStatus/icon?job=ci_linux&build=8156)](http://ci.ros2.org/job/ci_linux/8156/)
